pkg/apis/eventing/v1beta3/eventtype_types.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright 2021 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta3
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"knative.dev/pkg/apis"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	"knative.dev/pkg/kmeta"
+)
+
+// +genclient
+// +genreconciler
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// EventType represents a type of event that can be consumed from a Broker.
+type EventType struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the desired state of the EventType.
+	Spec EventTypeSpec `json:"spec,omitempty"`
+
+	// Status represents the current state of the EventType.
+	// This data may be out of date.
+	// +optional
+	Status EventTypeStatus `json:"status,omitempty"`
+}
+
+var (
+	// Check that EventType can be validated, can be defaulted, and has immutable fields.
+	_ apis.Validatable = (*EventType)(nil)
+	_ apis.Defaultable = (*EventType)(nil)
+
+	// Check that EventType can return its spec untyped.
+	_ apis.HasSpec = (*EventType)(nil)
+
+	_ runtime.Object = (*EventType)(nil)
+
+	// Check that we can create OwnerReferences to an EventType.
+	_ kmeta.OwnerRefable = (*EventType)(nil)
+
+	// Check that the type conforms to the duck Knative Resource shape.
+	_ duckv1.KRShaped = (*EventType)(nil)
+)
+
+type EventTypeSpec struct {
+	// Type represents the CloudEvents type. It is authoritative.
+	Type string `json:"type"`
+	// Source is a URI, it represents the CloudEvents source.
+	// +optional
+	Source *apis.URL `json:"source,omitempty"`
```

Comment:
```suggestion
	Source *apis.URL `json:"source"`
```

Reply:
this will go away anyways in this version.  the source will be required to be inside the map (see https://github.com/knative/eventing/issues/7265)